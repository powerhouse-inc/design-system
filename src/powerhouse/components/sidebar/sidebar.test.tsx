import { render, screen } from '@testing-library/react';

import { Sidebar, SidebarFooter, SidebarHeader, SidebarPanel } from '.';

describe('Sidebar Component', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(
            <Sidebar collapsed={false} maxWidth={'300px'} minWidth={'100px'}>
                <SidebarPanel>
                    <SidebarHeader />
                </SidebarPanel>
                <SidebarFooter />
            </Sidebar>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly', () => {
        render(
            <Sidebar
                data-testid="sidebar"
                collapsed={true}
                maxWidth={'300px'}
                minWidth={'100px'}
            />,
        );
        expect(screen.getByTestId('sidebar')).toBeInTheDocument();
    });

    it('should render expanded', () => {
        render(
            <Sidebar
                data-testid="sidebar"
                collapsed={false}
                maxWidth={'300px'}
                minWidth={'100px'}
            />,
        );
        expect(screen.getByTestId('sidebar')).toHaveStyle({ width: '300px' });
    });

    it('should render collapsed', () => {
        render(
            <Sidebar
                data-testid="sidebar"
                collapsed={true}
                maxWidth={'300px'}
                minWidth={'100px'}
            />,
        );
        expect(screen.getByTestId('sidebar')).toHaveStyle({ width: '100px' });
    });
});
