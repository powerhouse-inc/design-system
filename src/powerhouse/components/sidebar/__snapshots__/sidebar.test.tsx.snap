// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="group group/sidebar flex flex-col h-full bg-neutral-1 overflow-hidden shadow-[0px_33px_32px_-16px_rgba(0,0,0,0.10),0px_0px_16px_4px_rgba(0,0,0,0.04)]"
    style="width: 300px;"
  >
    <div
      class="flex-1 overflow-auto no-scrollbar transition-shadow"
      style="box-shadow: none;"
    >
      <div
        class="flex-shrink-0"
      />
    </div>
    <div
      class="flex-shrink-0"
    />
  </div>
</DocumentFragment>
`;
